### ISSUE TITLE
Shell Fails to compile with shared libs but separate shell app using published libs works fine

### ISSUE THREAD

**Issue by windbeneathyourwings** [angular-architects/module-federation-plugin#154]
I have been able to successfully work around a problem that I have ran into regarding compiling an app shell. I was able to do this by creating a separate app pulling in all the published libraries at which point sharing them in webpack config worked without compilation errors. That working app can be seen below.

https://github.com/ng-druid/ipe

Example page: https://ng-druid.github.io/tractorbeam-test-v3/

The libraries being used there are coming from this repo.

https://github.com/ng-druid/platform

However, I have been unable to successfully compile the ipe project inside that repo when modules shared. 

webpack.config.js
```typescript
const ModuleFederationPlugin = require("webpack/lib/container/ModuleFederationPlugin");
const mf = require("@angular-architects/module-federation/webpack");
const path = require("path");
const share = mf.share;

const sharedMappings = new mf.SharedMappings();
sharedMappings.register(
  path.join(__dirname, '../../tsconfig.json'),
  [/* mapped paths to share */]);

module.exports = {
  output: {
    uniqueName: "ipe",
    publicPath: "auto",
  },
  optimization: {
    runtimeChunk: false
  },   
  resolve: {
    alias: {
      ...sharedMappings.getAliases(),
    }
  },
  experiments: {
    outputModule: true
  },
  plugins: [
    new ModuleFederationPlugin({
        library: { type: "module" },

        shared: share({
          "@angular/core": { singleton: true, strictVersion: true, requiredVersion: 'auto' }, 
          "@angular/common": { singleton: true, strictVersion: true, requiredVersion: 'auto' }, 
          "@angular/common/http": { singleton: true, strictVersion: true, requiredVersion: 'auto' }, 
          "@angular/router": { singleton: true, strictVersion: true, requiredVersion: 'auto' },

          // Uncommenting this results in compilation errors for building project ipe
          "@ng-druid/utils": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },
          "@ng-druid/attributes": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },
          "@ng-druid/plugin": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },
          "@ng-druid/material": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },
          "@ng-druid/content": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },

          ...sharedMappings.getDescriptors()
        })
        
    }),
    sharedMappings.getPlugin()
  ],
};
```

Compilation results in a stream of errors indicating the shared modules can't be found even though they exist inside dist as built Angular libraries.

```text
Error: Module not found: Error: Can't resolve '@ng-druid/utils' in '/Users/hidden/projects/druid/dist/pages/fesm2020'
Error: Module not found: Error: Can't resolve '@ng-druid/content' in '/Users/hidden/projects/druid/dist/pages/fesm2020'
Error: Module not found: Error: Can't resolve '@ng-druid/plugin' in '/Users/hidden/projects/druid/dist/pages/fesm2020'
.... so on and so forth
```

Does anyone know what is going on here. It would be nice to be able to develop module federation features specifically in regards to sharing without needing publish all the libraries and use another project.

**Comment by dkhrunov** [comment]
> I have been able to successfully work around a problem that I have ran into regarding compiling an app shell. I was able to do this by creating a separate app pulling in all the published libraries at which point sharing them in webpack config worked without compilation errors. That working app can be seen below.
> 
> https://github.com/ng-druid/ipe
> 
> Example page: https://ng-druid.github.io/tractorbeam-test-v3/
> 
> The libraries being used there are coming from this repo.
> 
> https://github.com/ng-druid/platform
> 
> However, I have been unable to successfully compile the ipe project inside that repo when modules shared.
> 
> webpack.config.js
> 
> ```ts
> const ModuleFederationPlugin = require("webpack/lib/container/ModuleFederationPlugin");
> const mf = require("@angular-architects/module-federation/webpack");
> const path = require("path");
> const share = mf.share;
> 
> const sharedMappings = new mf.SharedMappings();
> sharedMappings.register(
>   path.join(__dirname, '../../tsconfig.json'),
>   [/* mapped paths to share */]);
> 
> module.exports = {
>   output: {
>     uniqueName: "ipe",
>     publicPath: "auto",
>   },
>   optimization: {
>     runtimeChunk: false
>   },   
>   resolve: {
>     alias: {
>       ...sharedMappings.getAliases(),
>     }
>   },
>   experiments: {
>     outputModule: true
>   },
>   plugins: [
>     new ModuleFederationPlugin({
>         library: { type: "module" },
> 
>         shared: share({
>           "@angular/core": { singleton: true, strictVersion: true, requiredVersion: 'auto' }, 
>           "@angular/common": { singleton: true, strictVersion: true, requiredVersion: 'auto' }, 
>           "@angular/common/http": { singleton: true, strictVersion: true, requiredVersion: 'auto' }, 
>           "@angular/router": { singleton: true, strictVersion: true, requiredVersion: 'auto' },
> 
>           // Uncommenting this results in compilation errors for building project ipe
>           "@ng-druid/utils": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },
>           "@ng-druid/attributes": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },
>           "@ng-druid/plugin": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },
>           "@ng-druid/material": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },
>           "@ng-druid/content": { singleton: true, strictVersion: true, requiredVersion: '0.0.24' },
> 
>           ...sharedMappings.getDescriptors()
>         })
>         
>     }),
>     sharedMappings.getPlugin()
>   ],
> };
> ```
> 
> Compilation results in a stream of errors indicating the shared modules can't be found even though they exist inside dist as built Angular libraries.
> 
> ```
> Error: Module not found: Error: Can't resolve '@ng-druid/utils' in '/Users/hidden/projects/druid/dist/pages/fesm2020'
> Error: Module not found: Error: Can't resolve '@ng-druid/content' in '/Users/hidden/projects/druid/dist/pages/fesm2020'
> Error: Module not found: Error: Can't resolve '@ng-druid/plugin' in '/Users/hidden/projects/druid/dist/pages/fesm2020'
> .... so on and so forth
> ```
> 
> Does anyone know what is going on here. It would be nice to be able to develop module federation features specifically in regards to sharing without needing publish all the libraries and use another project.

Hi, you define @ng-druid libs in each webpack.config.js files?

**Comment by windbeneathyourwings** [comment]
@dkhrunov

I have only defined @ng-druid/* libs inside the webpack.config.js file under the project being compiled. Do I need to add separate webpack.config.js files to each library. Is that what you are saying?

**Comment by dkhrunov** [comment]
@windbeneathyourwings Yes, you must define this lib inside each webpack.config.js in shared param of the ModuleFederationPlugin  in each Remote (aka MFE) app and each Host (aka shell) app, i think it should be help you!

Example:
<img width="427" alt="image" src="https://user-images.githubusercontent.com/25565058/161389056-ec977a89-b29b-4040-bf11-94e6f3f5e0f7.png">

**Comment by windbeneathyourwings** [comment]
@dkhrunov 

Oh I misunderstood.

Yes, those are declared in both the shell and micros.

The micro host project can be seen here which is a modified version of the workflow designer.

https://github.com/ng-druid/workflow-designer

**Comment by dkhrunov** [comment]
@windbeneathyourwings you dont define libs here - https://github.com/ng-druid/workflow-designer/blob/master/projects/shell/webpack.config.js

**Comment by windbeneathyourwings** [comment]
@dkhrunov 

Is that necessary since I'm not using that shell app. The shell app is the ipe project here.

https://github.com/ng-druid/platform/tree/master/projects/ipe

**Comment by dkhrunov** [comment]
@windbeneathyourwings 
Sorry, then i have not idea, my guess was wrong

**Comment by windbeneathyourwings** [comment]
It would be nice to find the root cause but isn't detrimental considering I have been able to successfully share the druid library services between the work flow designer when using an outside project that pulls in the published libs. Probably a good thing anyway having a separate standalone project anyway. It would be nice to easily test pages that require sharing services on the main project through.
